status/controller.go

```diff
@@ -114,25 +144,36 @@ func (c *Controller[T]) Reconcile(ctx context.Context, req reconcile.Request) (r
 	// time, and our likelyhood of observing this is much higher.
 	for _, condition := range currentConditions.List() {
 		observedCondition := observedConditions.Get(condition.Type)
-		if observedCondition == nil || observedCondition.GetStatus() == condition.GetStatus() {
+		if observedCondition.GetStatus() == condition.GetStatus() {
```

Comment:
Do we want to treat it as a transition if it's the same status but a different reason?

Reply:
No, I don't think that this should be considered a transition. We should consider a transition to be a status True/False change